tests/unit/ml_handlers/test_neuralforecast.py

```diff
@@ -140,3 +140,41 @@ def test_with_exog_vars(self, mock_handler):
 
         describe_features = self.run_sql("describe proj.model_exog_var.features")
         assert describe_features["exog_vars"][0] == ["exog_var_1"]
+
+    @patch("mindsdb.integrations.handlers.postgres_handler.Handler")
+    def test_hierarchical(self, mock_handler):
+        # create project
+        self.run_sql("create database proj")
+        df = pd.read_csv("tests/unit/ml_handlers/data/house_sales.csv")  # comes mindsdb docs forecast example
+        self.set_handler(mock_handler, name="pg", tables={"df": df})
+
+        self.run_sql(
+            """
+           create model proj.model_1_group
+           from pg (select * from df)
+           predict ma
+           order by saledate
+           group by type, bedrooms
+           horizon 4
+           window 8
+           using
+             engine='neuralforecast',
+             hierarchy=['type', 'bedrooms'],
+             train_time=0.01
+        """
+        )
+        self.wait_predictor("proj", "model_1_group")
+
+        # run predict
+        mindsdb_result_hier = self.run_sql(
```

Comment:
I'm wondering whether it makes sense to train a non-hierarchical model in this test, and check that results are different between them? Presumably there is _some_ impact in the forecast, at least. 

The big question in my mind would be whether NF is deterministic (i.e. are seeds fixed) so that we can guarantee two runs in a row for the same model config would yield equal results? If not, then this addition doesn't make sense. If yes, then I'd ask that you include this so we know there is an impact.

Reply:
Indeed - the issue is that NF is not deterministic (unlike SF, which is why the test for diffs works there).